crates/evm/traces/src/debug/sources.rs

```diff
@@ -0,0 +1,287 @@
+use eyre::{Context, Result};
+use foundry_common::compact_to_contract;
+use foundry_compilers::{
+    artifacts::{
+        sourcemap::{SourceElement, SourceMap},
+        Bytecode, ContractBytecodeSome, Libraries, Source,
+    },
+    multi::MultiCompilerLanguage,
+    Artifact, Compiler, ProjectCompileOutput,
+};
+use foundry_evm_core::utils::PcIcMap;
+use foundry_linking::Linker;
+use rayon::prelude::*;
+use rustc_hash::FxHashMap;
+use solang_parser::pt::SourceUnitPart;
+use std::{
+    collections::{BTreeMap, HashMap},
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+#[derive(Clone, Debug)]
+pub struct SourceData {
+    pub source: Arc<String>,
```

Comment:
can be `Arc<str>` I believe

Reply:
yea, but this `Arc<String>` is coming from `compilers::Source` and converting it into `Arc<str>` would be expensive I believe

should we change to to `Arc<str>` here? https://github.com/foundry-rs/compilers/blob/8ffdd72e9febd2c9a72d9363f439ce50b59a0db1/crates/artifacts/solc/src/sources.rs#L25